### PR TITLE
OrtResult: Fix calculation of excluded packages

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Sometimes as a result of an analyzer issue it can happen that a
dependency is not added to the list of packages in the analyzer result
but appears in the dependency tree. These packages were so far ignored
when calculating the exclusion state of packages. As a result
`isPackageExcluded()` returned always false for those packages which led
to consecutive errors, for example the `StatisticsCalculator` did not
count unresolved issues for those packages.

Fix the algorithm to take those packages into account. This also allows
to simplify the code in `isPackageExcluded()`.

### CODE REVIEW

**Comment by mnonnenmacher** [comment]
Closed in favor of #3433.

File: model/src/main/kotlin/OrtResult.kt [hunk 1]
```diff
@@ -112,33 +112,50 @@ data class OrtResult(
...
-                    val dependencies = scope.collectDependencies()
-                    includedPackages.addAll(dependencies)
+                    includedPackages += dependencies
                 }
             }
         }
 
-        val result = getPackages().associateBy(
+        val packages = getPackages()
+        val result = getPackages().associateByTo(
```
Comment (fviernau) [inline model/src/main/kotlin/OrtResult.kt:137]: getPackages().associateByTo( ->  packages.associateByTo(

File: model/src/main/kotlin/OrtResult.kt [hunk 2]
```diff
@@ -112,33 +112,50 @@ data class OrtResult(
...
+            mutableMapOf(),
             { curatedPackage -> curatedPackage.pkg.id },
             { curatedPackage ->
                 PackageEntry(
                     curatedPackage = curatedPackage,
-                    isExcluded = !includedPackages.contains(curatedPackage.pkg.id)
+                    isExcluded = curatedPackage.pkg.id !in includedPackages
```
Comment (fviernau) [inline model/src/main/kotlin/OrtResult.kt:143]: unrelated change

**Comment by sschuberth** [comment]
Can the remote branch be deleted?

File: model/src/main/kotlin/OrtResult.kt [hunk 3]
```diff
@@ -112,33 +112,50 @@ data class OrtResult(
...
+    private data class PackageEntry(val curatedPackage: CuratedPackage?, val isExcluded: Boolean)
 
     private val packages: Map<Identifier, PackageEntry> by lazy {
         log.info { "Computing excluded packages which may take a while..." }
 
+        val projects = getProjects()
+        val allPackages = mutableSetOf<Identifier>()
         val includedPackages = mutableSetOf<Identifier>()
-        getProjects().forEach { project ->
+        projects.forEach { project ->
```
Comment (fviernau) [inline model/src/main/kotlin/OrtResult.kt:123]: Consider now adding an empty line above.

File: model/src/main/kotlin/OrtResult.kt [hunk 4]
```diff
@@ -112,33 +112,50 @@ data class OrtResult(
...
             project.scopes.forEach { scope ->
                 val isScopeExcluded = getExcludes().isScopeExcluded(scope)
 
+                val dependencies = scope.collectDependencies()
```
Comment (fviernau) [inline model/src/main/kotlin/OrtResult.kt:127]: consider `scopeDependencies`
